//python -m grpc_tools.protoc -I protos --python_out=grpc_generated_files --grpc_python_out=grpc_generated_files protos/*.proto
syntax = "proto3";
package course;

message CourseTopic {
  string name = 1;
  string description = 2;
  int32 parent_id = 3;
  int32 locale_id = 4;
  bool is_active = 5;
  int32 sort = 6;
  int64 created_at = 7;
  int64 updated_at = 8;
  int64 deleted_at = 9;

}

message CreateCourseTopicRequest {
  string name = 1;
  string description = 2;
  int32 parent_id = 3;
  int32 locale_id = 4;
}


message CreateCourseTopicResponse {
  CourseTopic success_payload = 1;
  string error_payload = 2;
  int32 status_code = 3;
}




service CourseService {
  rpc CreateCourseTopic (CreateCourseTopicRequest) returns (CreateCourseTopicResponse);
}